app.js

```diff
@@ -16,18 +16,27 @@ async function cloneUsingSsh(action, settings) {
   const {
     overwrite, sshKey, extraArgs, saveCreds, username, password,
   } = action.params;
-  if (!path || !repo) { throw new Error("Both Clone Path and Repository are required parameters."); }
   const privateKey = sshKey || settings.sshKey;
-  // delete directory if already exists
-  if (overwrite) { await tryDelete(path); }
-  // parse args
-  let gitKey = null;
-  if (username && password) {
-    if (!repo.startsWith("https://")) {
-      throw new Error("Username and password authentication can be used only for repository URLs in HTTPS format.");
+
+  // validate parameters
+  if (repo.startsWith("https://")) {
+    if (!username || !password ) {
+      throw new Error("Both username and password are requried parameters for private repository URLs in HTTPS format.");
     }
     repo = `${repo.slice(0, 8)}${username}:${password}@${repo.slice(8)}`;
+  } else {
+    if (username || password) {
+      throw new Error("Parameters Username and Password are used only for repository URLs in HTTPS format.");
```

Comment:
should it actually be an error? It's redundant data, sure, but it doesn't exactly put the plugin in erroneous state, does it? Maybe simple `console.error` with a warning message would be enough?